src/slick.grid.ts

```diff
@@ -3806,7 +3806,7 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
 
     const frozenRowOffset = this.getFrozenRowOffset(row);
 
-    const rowHtml = `<div class="ui-widget-content ${rowCss}" style="top:${(this.getRowTop(row) - frozenRowOffset)}px">`;
+    const rowHtml = `<div class="ui-widget-content ${rowCss}" data-top="${(this.getRowTop(row) - frozenRowOffset)}px">`; //style="top:${(this.getRowTop(row) - frozenRowOffset)}px"
```

Comment:
why are you changing this to a `data-` attribute? is that to avoid the `unsafe-inline`? I'm not a big fan of this, it looks like you need to add an extra function to now deal with applying top margin, which again I'm not a big fan of... technically speaking we shouldn't have dynamic html string, I tried to replace all of them in my other open PR but I faced some issues with some of the grids so I didn't have a chance to finish it and I reverted my code but still we should use pure HTML Elements instead of data info, this will probably decrease the perf too

Reply:
yea its to dodge the unsafe-inline, not a fan either but the other options by dynamically creating the top as classes has other issues since you shouldnt double add a class and what happens if you add/delete items later then it should remove the classes too, tho performance wise this would be even worse.

Performance wise even with 500.000 rows it was non existent (i would have expected a bigger impact but below 0.09 ms) will do a few more tests to see how it hits performance also se we have actual numbers.

other option could be to make it an opt in option ex if this._options.nonce is set do it like this instead of normal way

Reply:
here are some actuel numbers
![image](https://github.com/6pac/SlickGrid/assets/137038231/033e3ec7-ff7d-443f-9bad-6c43df7eaaf7)
this is with 5.000.000 rows, doesnt seem like there is much of a difference between 500.000 and 5.000.000

Reply:
I now got code locally, not sure how to commit it without hitting this PR, where it doesn't use dynamic html strings but html elements instead, it has passed all the cypress tests except for 2 Firefox specific once (tho main repo also seems to fail on these) and new CSP example works without unsafe-inline

Reply:
Or should i just commit it to this PR`?

Reply:
> other option could be to make it an opt in option ex if this._options.nonce is set do it like this instead of normal way

yeah I was thinking if we should use a flag similar to what you added for DataView but yeah I guess using the `nonce` would work too. Even if using `data-top` is fast, it's still 2 tasks vs 1 task because you need an extra query selector. You might not see much difference because by default SlickGrid uses virtual scrolling, but if you use `autoHeight` (which will disable virtual scrolling), you should see a bigger difference but don't use a million rows because that will literally kill your browser (maybe 50,000). The virtual scroll will basically only create what you can see in the screen + 5 rows on top/bottom so yeah of course the query selector would be super fast 

We only run the Cypress tests with Chrome I think, which is the default, it should work in Firefox but I know Cypress first came out with Chrome at first and Firefox came after a few years. Anyway, just manually trying it in Firefox should be enough since I'm not sure Cypress tests are really stable with Firefox. The Cypress tests run on every PRs and we merge to master branch. What other change did you have locally?

Reply:
off note i have a change where i use html instead of dynamic html strings for this exact issue, should i commit it here? will properly add for now so it checks if the nonce value is set then it uses html instead of dynamic string, this way we we also dodge the data-top way.

For firefox test, alll tests generally go through the ones that fails seems to be the tests where it checks if it does something fast enough.
picture of the 2 failed tests
![image](https://github.com/6pac/SlickGrid/assets/137038231/d0eb6adc-f4ce-419b-a155-397b82f32d5d)

didnt actually know it was require-trusted-types-for 'script' that blocked innerHTML, tho it kinda makes sense since script-src only blocks script tags, ex if you attempt to append a script tag.

Reply:
I wouldn't trust the Cypress Firefox tests to be reliable especially for the auto-scroll which was written with specific delays in mind, if those delay don't fit for example if Firefox is slower for scrolling than other browsers then the test would fail... So I'm not surprised Firefox fails in that particular tests and maybe others, I always run Cypress against Chrome or Electron (which also uses Chromium), Firefox is slower in some areas so I don't test against it and it's also less supported in general by testing tools... in short make sure all Cypress tests are passing with Chrome, you can do quick visual tests in Firefox and that should be enough